test/preconditioner/batch_jacobi_kernels.cpp

```diff
@@ -0,0 +1,221 @@
+          d_b(BMVec::create(exec,
+                            gko::batch_dim<2>(nbatch, gko::dim<2>(nrows, 1)))),
+          ref_x(BMVec::create(
+              ref, gko::batch_dim<2>(nbatch, gko::dim<2>(nrows, 1)))),
+          d_x(BMVec::create(exec,
+                            gko::batch_dim<2>(nbatch, gko::dim<2>(nrows, 1))))
+    {
+        d_mtx->copy_from(ref_mtx.get());
+        d_b->copy_from(ref_b.get());
+        ref_scalar_jacobi_prec =
+            BJ::build().with_max_block_size(1u).on(ref)->generate(ref_mtx);
+        d_scalar_jacobi_prec =
+            BJ::build().with_max_block_size(1u).on(exec)->generate(d_mtx);
+        ref_block_jacobi_prec = BJ::build()
+                                    .with_max_block_size(max_blk_sz)
+                                    .on(ref)
+                                    ->generate(ref_mtx);
+
+        // TODO (before merging device kernels): Check if it is the same for
+        // other device kernels
+        // // so that the block pointers are exactly the same for ref and device
+        // const int* block_pointers_generated_by_ref =
+        //     ref_block_jacobi_prec->get_const_block_pointers();
+        // const auto num_blocks_generated_by_ref =
+        //     ref_block_jacobi_prec->get_num_blocks();
+
+        // gko::array<int> block_pointers_for_device(
+        //     this->exec, block_pointers_generated_by_ref,
+        //     block_pointers_generated_by_ref + num_blocks_generated_by_ref +
+        //     1);
+
+        d_block_jacobi_prec =
+            BJ::build()
+                .with_max_block_size(max_blk_sz)
+                // .with_block_pointers(block_pointers_for_device)
+                .on(exec)
+                ->generate(d_mtx);
+    }
+
+    template <typename MatrixType>
+    gko::test::LinearSystem<MatrixType> setup_linsys_and_solver(
+        std::shared_ptr<const MatrixType> mat, const int num_rhs,
+        const real_type tol, const int max_iters)
+    {
+        auto executor = exec;
+        solve_lambda = [executor](const Settings settings,
+                                  const gko::batch::BatchLinOp* prec,
+                                  const Mtx* mtx, const MVec* b, MVec* x,
+                                  LogData& log_data) {
+            gko::kernels::EXEC_NAMESPACE::batch_bicgstab::apply<
+                typename Mtx::value_type>(executor, settings, mtx, prec, b, x,
+                                          log_data);
+        };
+        solver_settings = Settings{max_iters, tol,
+                                   gko::batch::stop::tolerance_type::relative};
+        precond_solver_factory =
+            solver_type::build()
+                .with_max_iterations(max_iters)
+                .with_tolerance(tol)
+                .with_tolerance_type(gko::batch::stop::tolerance_type::relative)
+                .with_preconditioner(
+                    precond_type::build().with_max_block_size(2u))
```

Comment:
The preconditioner does not check the scalar. I am a bit confused at the `d_scalar_block_prec` only checks the generated but here use another one